### PR TITLE
Added outlined-pixel pass example

### PR DESCRIPTION
**Pull request by KodyJKing**
This is a pixelization effect with outlines that add a bit of a pixel art feel.

https://www.youtube.com/watch?v=-WHW48SlvsI

### CODE REVIEW

**Comment by WestLangley** [comment]
Live link
https://raw.githack.com/KodyJKing/three.js/outlined-pixel-example/examples/webgl_postprocessing_pixel_outlines.html

**Comment by Mugen87** [comment]
There is already: https://threejs.org/examples/webgl_postprocessing_pixel

Would this PR be some sort of improvement over the existing approach?

**Comment by gkjohnson** [comment]
I think the example is much more compelling than the existing one personally especially for people who might be looking to make retro-style games. The addition of normal and depth edge sharpening is really nice and gives a more stylized video game sprite-like effect -- it looks nice.

The down side is that it requires a lot more shader texture sampling to achieve those edges. If performance impact is a concern then maybe the effect could be adjusted so no depth or normal sampling is done if either corresponding edge strength parameter equals zero. Otherwise I'd think this would be a neat replacement for the existing pixel effect.

**Comment by Mugen87** [comment]
As a next step, I suggest you replace the existing `webgl_postprocessing_pixel` example with your code. Since you have implemented the effect as a pass, `PixelShader` can be removed.

**Comment by haroldiedema** [comment]
I just wanted to say that this is bloody awesome and runs smooth as butter on my 'el-cheapo' Galaxy A50. Very nicely done! ❤️❤️

File: examples/jsm/postprocessing/RenderPixelatedPass.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+				resolution: {
+					value: new Vector4(
+						this.renderResolution.x,
+						this.renderResolution.y,
+						1 / this.renderResolution.x,
+						1 / this.renderResolution.y,
+					)
+				},
+				normalEdgeStrenth: { value: 0 },
+				depthEdgeStrenth: { value: 0 }
```
Comment (haroldiedema) [inline examples/jsm/postprocessing/RenderPixelatedPass.js:117]: Typo: strenth = strength.

File: examples/files.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
- {
```
Comment (Mugen87) [inline examples/files.json:1]: Please try to avoid such formatting changes. They just produce unwanted diffs.

File: examples/files.json [hunk 3]
```diff
@@ -408,4 +408,4 @@
 		"webgl_pmrem_test",
 		"misc_uv_tests"
 	]
-}
+}
```
Comment (Mugen87) [inline examples/files.json:411]: Same here.

**Comment by KodyJKing** [comment]
@Mugen87 I replaced `webgl_postprocessing_pixel ` and removed the `PixelShader`. However, I'm thinking `PixelShader` still has a use case because my `RenderPixelatedPass` handles the initial render of the scene and expects to be the first pass in the effect composer. If someone wanted to render the scene and apply various effects before pixelization, my pass wouldn't work for them.

**Comment by Mugen87** [comment]
I understand but I still think a single pixel shader implementation is sufficient. We have to maintain all this code and thus make compromises. 

Besides, a basic pixel effect is not hard to implement. Most users should figure out by themselves to implement it.

**Review by Mugen87 — APPROVED** [review]

**Comment by drcmda** [comment]
@Mugen87 @KodyJKing this looks 😍, but before this gets into 140 ..., this is not a correct import:

```jsx
} from '../../../build/three.module.js';
```

this will double threejs again by importing two namespaces. it should just be `three`.

**Comment by Mugen87** [comment]
Closing in favor of #24873.

**Comment by DefinitelyMaybe** [comment]
Hey @KodyJKing !

Loving the pixel effect! thank you so much for working on this :1st_place_medal:

**Comment by KodyJKing** [comment]
Oh, I haven't checked in on this in a while. Thanks @Mugen87 for getting it ready to merge.

**Pull request closed without merging**